docs/building-in-chunks/batch-propose.md

```diff
@@ -0,0 +1,73 @@
+
+On the other hand, if we consider 2400 bytes per tx effect, which is the size of a transfer with fpc support with all notes posted on-chain, then each blob can carry about 50 txs only. Assuming 10 TPS, this means we fill 2 blobs per L1 slot, so checkpoints cannot be longer than 3 L1 slots (36s).
+
+**TLDR**:
+
+- At 10 TPS with 1200 byte txs, checkpoints happen every 72s, block time is 12s (good)
+- At 6 TPS with 300 byte txs, block time is 72s (bad UX)
+- At 10 TPS with 2400 byte txs, checkpoints happen every 36s, block time is 6s (worst cost)
+
+### Gas savings
+
+Following are the actions we do in `Rollup.propose` as of today, and whether they can be done once per batch or would still have to be repeated once per block. Gas costs are estimated from [this doc](https://hackmd.io/j-H-8k8dRYSAjXCpzk4WPg).
+
+| Action                         | Per block | Per batch | Gas cost |
+| ------------------------------ | --------- | --------- | -------- |
+| Base tx cost                   |           | x         | 21k      |
+| Post blob inputs               | x         |           | 20k      |
+| Setup epoch                    |           | x         | 3k       |
+| Update L1 gas fee oracle       |           | x         | 8k       |
+| Validate blob commitments      | x         |           | 8k       |
+| Compute base fee               |           | x         | 22k      |
+| Validate block header          | x         |           | 20k      |
+| Post and validate attestations |           | x         | 200k     |
+| Update pending chain tip       |           | x         | 3k       |
+| Store archive root             | x         |           | 22k      |
+| Store block log                | x         |           | 30k      |
+| Consume inbox                  | x         |           | 10k      |
+| Push messages to outbox        | x         |           | 9k       |
+| Emit log                       | x         |           | 3k       |
+| Unaccounted for                |           | x         | 20k      |
+
+The total gas cost from `propose` today is then 400k gas every 36s, or 11k gas per second. Under this new model, the cost per block is `20 + 8 + 20 + 22 + 30 + 10 + 9 + 3` or 122k gas, and the cost per batch is `21 + 3 + 8 + 22 + 200 + 3 + 20` or 277k gas. Assuming 72s checkpoints:
+
+- With 72s checkpoints and 12s blocks (best for UX), total gas cost per checkpoint is 1M, or 14k gas per second (worse than status quo)
+- Decreasing block frequency to once per 36s (same UX as today), total gas cost per checkpoint is 521k gas, or 7k gas per second (33% better than status quo)
+- Halving checkpoints to 36s to support 2400-byte txs, with one block every 6s (best for UX), total gas cost is 1M, or 28k gas per second (much worse than status quo)
```

Comment:
I don't understand this. Why would checkpoints be every 36s here, and why is the data so high, what needs 4 notes and nullifiers with data on chain? At most a FPC for a private transaction needs 3 notes, and one is transient.

Reply:
> Why would checkpoints be every 36s here

This is assuming we consume 24kb/s in blob data and there are only 6 usable blobs per L1 blocks.

> why is the data so high,

Can I defer to you @iAmMichaelConnor on this?